src/actions/action_menu.ts

```diff
@@ -0,0 +1,299 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  ASTNode,
+  Connection,
+  ContextMenu,
+  ContextMenuRegistry,
+  ShortcutRegistry,
+  comments,
+  utils as BlocklyUtils,
+  WidgetDiv,
+} from 'blockly';
+import * as Constants from '../constants';
+import type {BlockSvg, WorkspaceSvg} from 'blockly';
+import {Navigation} from '../navigation';
+
+const KeyCodes = BlocklyUtils.KeyCodes;
+const createSerializedKey = ShortcutRegistry.registry.createSerializedKey.bind(
+  ShortcutRegistry.registry,
+);
+
+interface Scope {
```

Comment:
Perhpas this should be exported so it can be imported by the menu items that now support connection contexts (which IIRC currenlty have their own copy of this type).